packages/shared/lib/services/hmac.service.ts

```diff
@@ -21,8 +21,9 @@ class HmacService {
         return key;
     }
 
-    async verify(expectedDigest: string, id: number, ...values: string[]): Promise<boolean> {
-        const actualDigest = await this.digest(id, ...values);
+    async verify(expectedDigest: string, id: number, ...values: (string | undefined)[]): Promise<boolean> {
+        const definedValues: string[] = values.flatMap((v) => (v === undefined ? [] : [v]));
```

Comment:
filter(Boolean) would not work here?

Reply:
you are right, that should work. let me check

Reply:
it does work but it will require to cast to `string[]` because filter doesn't change the type of values